OMCompiler/Compiler/Util/Flags.mo

```diff
@@ -1503,15 +1507,15 @@ public function getConfigValue
   input ConfigFlag inFlag;
   output FlagData outValue;
 protected
-  array<FlagData> config_flags;
-  Integer index;
-  Flag flags;
-  String name;
+  UnorderedMap<String, FlagData> config_flags;
 algorithm
-  CONFIG_FLAG(name = name, index = index) := inFlag;
-  flags := getFlags();
-  FLAGS(configFlags = config_flags) := flags;
-  outValue := arrayGet(config_flags, index);
+  FLAGS(configFlags = config_flags) := getFlags();
+  if UnorderedMap.contains(inFlag.name, config_flags) then
+    outValue := UnorderedMap.getSafe(inFlag.name, config_flags, sourceInfo());
+  else
+    // this should not happen, but in case its not set, just use default
+    outValue := inFlag.defaultValue;
+  end if;
```

Comment:
It would be better to use `UnorderedMap.get` and then check if it's `SOME` or not, instead of doing `contains` followed by `getSafe` which effectively means looking up (and hashing) the key twice.

Reply:
Thanks, should probably do that in multiple places!